DuckDuckGo/AppConfigurationFetch.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  AppBackgroundFetch.swift
+//  DuckDuckGo
+//
+//  Copyright © 2018 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Core
+
+public typealias AppConfigurationCompletion = (Bool) -> Void
+
+class AppConfigurationFetch {
+    
+    private lazy var statisticsStore: StatisticsStore = StatisticsUserDefaults()
+
+    func start(completion: AppConfigurationCompletion?) {
+
+        DispatchQueue.global(qos: .background).async {
+
+            var newData = false
+            let semaphore = DispatchSemaphore(value: 0)
+            
+            self.sendHttpsSummaryPixel { newHttpsData in
+                newData = newData || newHttpsData
+                semaphore.signal()
+            }
+            semaphore.wait()
+
+            ContentBlockerLoader().start { newContentBlockingData in
+                newData = newData || newContentBlockingData
+                semaphore.signal()
+            }
+            semaphore.wait()
+            
+            completion?(newData)
+        }
+    }
+    
+    private func sendHttpsSummaryPixel(completion: @escaping (Bool) -> Void) {
+            
```

Comment:
In this case we're sending data rather than receiving it but it still feels valid to tell apple that this sync "needed to happen" versus not. I understand that we want to kill this entirely one day, for now, I feel it's in keeping with the existing pattern in the app.